src/main/java/cubicchunks/server/ServerCubeCache.java

```diff
@@ -279,9 +274,11 @@ public Cube getCube(int cubeX, int cubeY, int cubeZ, @Nonnull Requirement req) {
 	 */
 	private void onCubeLoaded(@Nullable Cube cube, @Nonnull Column column) {
 		if(cube != null) {
-			column.addCube(cube);
 			cubeMap.put(cube); // cache the Cube
-			cube.onLoad();             // init the Cube
+			if(!column.getLoadedCubes().contains(cube)) {
+				column.addCube(cube);
+				cube.onLoad(); // init the Cube
+			}
```

Comment:
Could this potentially lead to the column having a different version of the cube than the cubeMap?